app/models/concerns/isolation_symptom_onset_validator.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# Validates that when in Isolation, either:
+# - symptom_onset is set to a value
+# - the patient has at least one positive lab result
+class IsolationSymptomOnsetValidator < ActiveModel::Validator
+  include ValidationHelper
+  def validate(record)
+    return unless record.isolation &&
+                  record.symptom_onset.nil? &&
+                  record.laboratories.find { |lab| lab.result == 'positive' && !lab.specimen_collection.blank? }.nil?
```

Comment:
This might be a more efficient query (not tested yet):
```suggestion
                  record.laboratories.exists?("laboratories.result = 'positive' AND laboratories.specimen_collection IS NOT NULL")
```

Reply:
This query isn't working for me. This is the SQL it's generating for me doesn't seem to include the conditions:
```
SELECT 1 AS one FROM `laboratories` WHERE `laboratories`.`patient_id` = 10 AND `laboratories`.`id` = NULL LIMIT 1
```

Reply:
In this case, I'm not sure we really need to query, since this validator is only run on create, the patient and its corresponding labs are not yet saved in the database.